### PR TITLE
Adding `is_valid_resource_name` helper.

### PR DESCRIPTION
**Pull request by marstr**
Definition of ARM Resource name found here: https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/resource-api-reference.md#arguments-for-crud-on-resource

related to Azure/azure-cli#6343

### CODE REVIEW

**Comment by marstr** [comment]
Looks like CI is failing for pre-existing conditions. :/ I can try to fix that in a separate PR.

File: tests/test_tools.py [hunk 1]
```diff
@@ -475,6 +475,27 @@ def test_resource_parse(self):
...
+        invalid_names = [
+            '',
+            'knights/ni',
+            'this&that',
+            'i<3you',
+            ''.join('a' for _ in range(261))
+        ]
+
+        for test in invalid_names:
+            self.assertFalse(is_valid_resource_name(test), '"{}" should be invalid'.format(test))
```
Comment (lmazuel) [inline tests/test_tools.py]:
```python
assert not is_valid_resource_name(test)
```
You don't need the string, pytest will do it for you
https://docs.pytest.org/en/latest/assert.html

File: tests/test_tools.py [hunk 2]
```diff
@@ -475,6 +475,27 @@ def test_resource_parse(self):
...
+
+        valid_names = [
+            'abc-123',
+            ' ',  # no one said it had to be a good resource name.
+            ''.join('a' for _ in range(260))
+        ]
+
+        for test in valid_names:
+            self.assertTrue(is_valid_resource_name(test), '"{}" should be valid'.format(test))
```
Comment (lmazuel) [inline tests/test_tools.py]:
```python
assert is_valid_resource_name(test)
```
You don't need the string, pytest will do it for you
https://docs.pytest.org/en/latest/assert.html

Comment (marstr) [reply tests/test_tools.py]: Fixed in 1e43d02

File: tests/test_tools.py (hunk 1 above)
Comment (marstr) [reply tests/test_tools.py]: Fixed in 1e43d02

**Pull request merged**